scripts/get_das_info.py

```diff
@@ -22,10 +25,105 @@ def convert_to_desired_structure(data: dict) -> str:
     ],
     n_files={data['nfiles']},
     n_events={data['nevents']},
+)
+"""
+
+
+identifier_map = {
+    "_TuneCP5Down_": "tune_down",
+    "_TuneCP5Up_": "tune_up",
+    "_TuneCP5CR1_": "cr_1",
+    "_TuneCP5CR2_": "cr_2",
+    "_Hdamp-158_": "hdamp_down",
+    "_Hdamp-418_": "hdamp_up",
+    "_MT-171p5_": "mtop_down",
+    "_MT-173p5_": "mtop_up",
+    # dataset types that I have no use for but want to keep anyways
+    "_MT-166p5_": "xxx",
+    "_MT-169p5_": "xxx",
+    "_MT-175p5_": "xxx",
+    "_MT-178p5_": "xxx",
+    "_DS_TuneCP5_": "xxx",
+    "_TuneCP5_ERDOn_": "xxx",
+    "_TuneCH3_": "xxx",
+    # nominal entry as the last one such that other dataset types get priority
+    "_TuneCP5_": "nominal",
+}
+
+
+def convert_for_top_queries(data: dict) -> str:
+    """
+    Function that converts dataset info into either an order Datset for nominal datasets
+    or to a DatasetInfo for variations of datasets such as tune or mtop.
+
+    Exemplary usage:
+    python get_das_info.py -f convert_for_top_queries -d "/TTtoLNu2Q*/Run3Summer22EENanoAODv12-130X_*/NANOAODSIM"
+    """
+    dataset_type = None
+
+    for identifier in identifier_map:
+        if identifier in data["name"]:
+            dataset_type = identifier_map[identifier]
+            break
+
+    if not dataset_type:
+        return f"""
+        #####
+        #####ERROR! Did not manage to determine type of dataset {data['name']}
+        #####
+        """
+
+    if dataset_type == "nominal":
+        return f"""cpn.add_dataset(
+    name="PLACEHOLDER",
+    id={data['dataset_id']},
+    processes=[procs.PLACEHOLDER],
+    info=dict(
+        nominal=DatasetInfo(
+            keys=[
+                "{data['name']}",  # noqa
+            ],
+            n_files={data['nfiles']},
+            n_events={data['nevents']},
+        ),
+    ),
 )"""
+    elif dataset_type == "xxx":
+        # comment out this dataset
+        return f"""        # {dataset_type}=DatasetInfo(
+        #     keys=[
+        #         "{data['name']}",  # noqa
+        #     ],
+        #     n_files={data['nfiles']},
+        #     n_events={data['nevents']},
+        # ),"""
+    else:
+        # some known variation of the dataset
+        return f"""        {dataset_type}=DatasetInfo(
+            keys=[
+                "{data['name']}",  # noqa
+            ],
+            n_files={data['nfiles']},
+            n_events={data['nevents']},
+        ),"""
+
+
+convert_functions = {
+    "convert_to_desired_structure": convert_to_desired_structure,
+    "convert_for_top_queries": convert_for_top_queries,
```

Comment:
The function names are quite verbose, maybe they can be shortened to e.g. `default` and `top_datasets` or similar?